dotnet/src/SemanticKernel.Core/Kernel.cs

```diff
@@ -228,8 +228,6 @@ public async Task<KernelResult> RunAsync(ContextVariables variables, Cancellatio
 
                 functionResult = await skFunction.InvokeAsync(context, cancellationToken: cancellationToken).ConfigureAwait(false);
 
-                context = functionResult.Context;
```

Comment:
This is breaking change with unknown outcomes. That's the reason why property `Context` still exists in `FunctionResult` model internally. Because, some functions may update context inside (since context is mutable at the moment), but some of them can return new cloned context instance with updated parameters (that's probably the way how it should be), which means that next function in pipeline won't receive updated variables if we remove this line.

We should investigate and test all available scenarios to understand if we can safely do this change and if we need to do it at all.